### PR TITLE
make tempDir optional

### PR DESCRIPTION
**Pull request by ikitiki**
and rename it to stagingDir to make naming more homogeneous;
simplify SyncFileAndDirectory;

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -21,6 +20,7 @@ type Config struct {
 	InitialBasebackup                      bool           `yaml:"initialBasebackup"`
 	SendStatusOnCommit                     bool           `yaml:"sendStatusOnCommit"`
 	Fsync                                  bool           `yaml:"fsync"`
+	StagingDir                             string         `yaml:"tempDir"`
```
[the remark this answers is not in the document]
Comment (ikitiki) [reply pkg/config/config.go]: I didn't want to break current config.

File: pkg/logicalbackup/logicalbackup.go [hunk 2]
```diff
@@ -123,9 +123,9 @@ func New(ctx context.Context, stopCh chan struct{}, cfg *config.Config) (*Logica
 	mux.Handle("/debug/pprof/trace", http.HandlerFunc(pprof.Trace))
 
 	lb := &LogicalBackup{
-		ctx:                    ctx,
-		stopCh:                 stopCh,
-		dbCfg:                  pgxConn,
+		ctx:    ctx,
```
[the remark this answers is not in the document]
Comment (ikitiki) [reply pkg/logicalbackup/logicalbackup.go]: for some reason `go fmt` reformatted it

File: pkg/config/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexeyklyukin) [reply pkg/config/config.go]: IMO it's not production-ready yet, so changing a config in favor of better names for options is absolutely fine.

File: pkg/tablebackup/tablebackup.go [hunk 3]
```diff
@@ -24,10 +24,12 @@ import (
 )
 
 const (
+	// TODO: move those constants somewhere else
+	DeltasDirName      = "deltas"
```
Comment (alexeyklyukin) [inline pkg/tablebackup/tablebackup.go]: Do we really need to export those constants?

File: pkg/tablebackup/tablebackup.go [hunk 4]
```diff
@@ -477,7 +503,13 @@ func (t *TableBackup) hasRows() (bool, error) {
 func (t *TableBackup) Truncate() error {
```
Comment (alexeyklyukin) [inline pkg/tablebackup/tablebackup.go]: It looks like this method is not used anymore.

File: pkg/utils/utils.go [hunk 5]
```diff
@@ -4,20 +4,16 @@ import (
...
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/ikitiki/logical_backup/pkg/dbutils"
-	"github.com/ikitiki/logical_backup/pkg/message"
 )
 
-func TableDir(tbl message.NamespacedName, oid dbutils.Oid) string {
-	if oid == dbutils.InvalidOid {
```
Comment (alexeyklyukin) [inline pkg/utils/utils.go]: Please, don't kill this check yet.

Comment (ikitiki) [reply pkg/utils/utils.go]: I'm going to kill it in the next PR

File: pkg/tablebackup/tablebackup.go (hunk 4 above)
Comment (ikitiki) [reply pkg/tablebackup/tablebackup.go]: removed

File: pkg/tablebackup/tablebackup.go (hunk 3 above)
Comment (ikitiki) [reply pkg/tablebackup/tablebackup.go]: it'll be used in the next PR, but I'll revert it as for now

**Pull request merged**